### ISSUE TITLE
[Bug]: How to train on multi gpu?

### ISSUE THREAD

**Issue by eric-debug666** [speechbrain/speechbrain#1972]
### Describe the bug

I attempted to train the conformer model from scratch for multi-gpu when runnung
 python3 train_asr.py hparams/conformer_m.yaml --data_folder=/dataSSD/data_aishell/ --data_parallel_backend
then, I encountered the following error
![image](https://user-images.githubusercontent.com/62170105/236642254-6ef6d3fd-faa2-423e-a74e-7328ac348f67.png)

### Expected behaviour

![image](https://user-images.githubusercontent.com/62170105/236642283-edb88c5e-591b-420d-9abd-51777321eed5.png)

**Comment by Adel-Moumen** [comment]
Hi, 

Please switch to Distributed Data Parallel. You can have a look on how to use it with SpeechBrain here: https://speechbrain.readthedocs.io/en/latest/multigpu.html

**Comment by Adel-Moumen** [comment]
Hey, any news about this issue pls ? thanks

**Comment by eric-debug666** [comment]
Hey, thanks the suggestion from @Adel-Moumen.
However, if I switched the strategy to DDP, the result is worse than the result I train on single GPU.
Is there anything wromg from DDP?

**Comment by Adel-Moumen** [comment]
Hmmmm I think it should be slightly the same results. What are the results you are getting so far ?

**Comment by eric-debug666** [comment]
I attempted to train a conformer model(from the recipes of LibriSpeech) for the dataset of AISHELL-1. The result trained on single GPU (global batch size : 128) is 6.2%/6.71% (CER). If I train on multi gpu applied the strategy of DDP (global batch size : 256), the result is 6.74%/7.56% (CER). However, the result from the paper is 4.39%/4.89%. I have tried my best to train in same parameter, but there is still a gap between them. 
@Adel-Moumen Is there any suggestions?
Thanks a lot

**Comment by Adel-Moumen** [comment]
You should definitely use the Transducer Conformer from this PR https://github.com/speechbrain/speechbrain/pull/1782 not merged yet but it works.

**Comment by Adel-Moumen** [comment]
Hey @eric-debug666, any news regarding this issue please?

**Comment by eric-debug666** [comment]
@Adel-Moumen Thanks for your suggestion. I have trained on the conformer transducer model. Due to the insufficient memory capacity, I didn't apply the dynamic batch size. I set batch size of 8, and gradient accumulation factor as 1(the original setting). The result seemed to be something wrong. The result of third epoch is CER of 98.24(%) and WER of 100(%). In my mind, it indicates that the global batch size is not enough(the command suggests that global batch size should be equal or larger than 128). 
@Adel-Moumen  Is there any suggestions for the fixed batch size? Does the batch size of 16 and the accumulation factor of 8 (16*8*1=128) work?

**Comment by Adel-Moumen** [comment]
Do you know if there is any workaround regarding Conformer and batch size @TParcollet ? Thanks.

**Comment by TParcollet** [comment]
Hi, Dynamic batching must be used when you have little amount of VRAM. It's the best way of adjusting to difficult scenarios as you can tailor with respect to the max number of second per batch === very precise. In most of my experiences with Conformer and libri, I try to have around 1000s as a total batch.

**Comment by Adel-Moumen** [comment]
Hey @eric-debug666, do you have any news regarding this issue please? Thanks. :)

**Comment by Adel-Moumen** [comment]
Hello, 

It's been quite some time without any updates or news on this issue. Consequently, I'm taking the step to close it. Feel free to reopen it if you are still facing the issue. 

Best,
Adel
